src/ccalls.jl

```diff
@@ -3,12 +3,32 @@
 import Base: getproperty, hasproperty, setproperty!, propertynames
 
 function RAW_GAP_TO_JULIA(ptr::Ptr{Cvoid})::Any
+    # convert immediate ints and FFEs directly, to void (un)boxing
+#    as_int = convert(Int, ptr)
+#    if as_int & 1 == 1
+#        return as_int >> 2
+#    elseif as_int & 2 == 2
+#        return reinterpret(FFE, as_int)
+#    end
     return ccall(:julia_gap, Any, (Ptr{Cvoid},), ptr)
 end
 
 function RAW_JULIA_TO_GAP(val::Any)::Ptr{Cvoid}
     return ccall(:gap_julia, Ptr{Cvoid}, (Any,), val)
 end
+RAW_JULIA_TO_GAP(x::FFE) = reinterpret(Ptr{Cvoid}, x)
+RAW_JULIA_TO_GAP(x::GapObj) = pointer_from_objref(x)
+function RAW_JULIA_TO_GAP(x::Int64)
+    # if it fits into a GAP immediate integer, convert x to Int64
+    if x in -1<<60:(1<<60-1)
```

Comment:
```suggestion
    if x in -one(Int64)<<60:(one(Int64)<<60-1)
```

Reply:
What's the difference?

Reply:
`1` is of type `Int`, so `1<<60` overflows on 32-bit machines (which I assumed are supported as the `x` argument is explicitely typed as `Int64` and not `Int`).

Reply:
Actually 32bit is not supported at all